group/views.py

```diff
@@ -20,3 +23,100 @@ def create_group(request):
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
     return None
+
+
+@api_view(['PATCH'])
+@permission_classes([IsAuthenticated])
+def add_users(request):
+    """Add user(s) to a group"""
+    if request.method == 'PATCH':
+        data = request.data.copy()
+        group_id = request.data.get("group_id")
+
+        data.pop('creator', None)
+        data.pop('name', None)
+        try:
+            group = Group.objects.get(id=group_id)
+            if request.user != group.creator:
+                return Response({
+                                "error": "Only group creator can add members"},
+                                status=status.HTTP_403_FORBIDDEN)
+            errors = []
+            for member in group.members.all():
+                if member.id not in data['members']:
+                    data['members'].append(member.id)
+            for member in data['members']:
+                try:
+                    user = User.objects.get(id=member)
+                    if user.id not in group.members.all()\
+                            .values_list('id', flat=True):
+                        group.members.add(user)
+                except User.DoesNotExist:
+                    errors.append(f"User with ID {member} is not found")
+            serializer = GroupSerializer(group, data=data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            if errors:
+                return Response({"error": errors},
+                                status=status.HTTP_404_NOT_FOUND)
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+        except Group.DoesNotExist:
+            return Response({"error": "Group not found"},
+                            status=status.HTTP_404_NOT_FOUND)
+    return None
+
+
+@api_view(['DELETE'])
+@permission_classes([IsAuthenticated])
+def delete_group(request):
+    """Delete a group"""
+    if request.method == 'DELETE':
+        group_id = request.data.get("group_id")
+        try:
+            group = Group.objects.get(id=group_id)
+            if request.user != group.creator:
+                return Response({"error": "Only the group creator can delete"})
+            group.delete()
+            return Response({"message": "Group deleted successfully"},
+                            status=status.HTTP_204_NO_CONTENT)
+        except Group.DoesNotExist:
+            return Response({"error": "Group not found"},
+                            status=status.HTTP_404_NOT_FOUND)
+    return None
+
+
+@api_view(['DELETE'])
+@permission_classes([IsAuthenticated])
+def remove_user(request):
+    """Remove a user(s) from a group"""
+    if request.method == 'DELETE':
+        group_id = request.data.get("group_id")
+        user_id = request.data.get("user_id")
+        try:
```

Comment:
You can actually have one try block here, and two expect blocks i.e (User.DoesNotExist and Group.DoesNotExist). See such examples here https://docs.python.org/3/tutorial/errors.html

Reply:
Generally, I will advise you use `class based` views for DRF